platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/GradleEnterprisePluginServices.java

```diff
@@ -44,10 +47,8 @@ public void registerGradleServices(ServiceRegistration registration) {
         registration.add(DefaultGradleEnterprisePluginAdapter.class);
         registration.add(DefaultGradleEnterprisePluginBackgroundJobExecutors.class);
         registration.add(DefaultGradleEnterprisePluginBuildState.class);
```

Comment:
❓ This service is also annotated with the BuildTree scope. Shall we move it to the correct registration as well?

Reply:
I trusted the annotation and moved it as well. Let's see what tests show. @ldaley , does this make sense or is the annotation wrong?

Reply:
Upon closer inspection, it seems that the annotation on the class is wrong.

- It uses (gets injected with) a `GradleInternal` instance, which is [`Build` scoped](https://github.com/gradle/gradle/blob/995ce143d23bf82458271916ffcb68de8708aaec/subprojects/base-annotations/src/main/java/org/gradle/internal/service/scopes/Scopes.java#L57) 
- It also uses a `BuildInvocationScopeId`, which is `Gradle` scoped that is "stricter" than `Build` scoped.

This leads me to believe that the class should have been annotated with the `Gradle` scope itself. And then its registration via `registerGradleServices` was correct.

cc @wolfs